src/Twig/Extension/SonataAdminExtension.php

```diff
@@ -114,11 +124,52 @@ public function __construct(
             }
         }
 
+        // NEXT_MAJOR: Remove this block.
+        if (!$propertyAccessorOrSecurityChecker instanceof PropertyAccessorInterface
+            && !$propertyAccessorOrSecurityChecker instanceof AuthorizationCheckerInterface
+            && null !== $propertyAccessorOrSecurityChecker
+        ) {
+            throw new \TypeError(sprintf(
+                'Argument 5 must be an instance of "%s" or "%s" or null, "%s given"',
+                PropertyAccessorInterface::class,
+                AuthorizationCheckerInterface::class,
+                \is_object($propertyAccessorOrSecurityChecker) ? \get_class($propertyAccessorOrSecurityChecker) : \gettype($propertyAccessorOrSecurityChecker)
+            ));
+        }
+
+        // NEXT_MAJOR: Remove this block and extract the else part outside.
+        if ($propertyAccessorOrSecurityChecker instanceof AuthorizationCheckerInterface) {
+            @trigger_error(sprintf(
+                'Passing an instance of "%s" the argument 5 for "%s()" is deprecated since sonata-project/admin-bundle'
+                .' 3.x and will throw a \TypeError in version 4.0. You MUST pass an instance of "%s" instead and pass'
+                .' the instance of "%s" as 6 argument.',
```

Comment:
```suggestion
                'Passing an instance of "%s" as argument 5 for "%s()" is deprecated since sonata-project/admin-bundle'
                .' 3.x and will throw a \TypeError error in version 4.0. You MUST pass an instance of "%s" instead and pass'
                .' an instance of "%s" as argument 6.',
```